xtuner/engine/hooks/evaluate_chat_hook.py

```diff
@@ -180,10 +198,18 @@ def after_train_iter(self,
                          data_batch=None,
                          outputs=None) -> None:
         if (self.every_n_iters is None or batch_idx == 0
-                or batch_idx % self.every_n_iters != 0):
+                or (batch_idx + 1) % self.every_n_iters != 0):
```

Comment:
`batch_idx + 1` will cause the execution iters of evaluating generation and saving ckpt to be inconsistent.

Reply:
batch_idx = runner.iter = (real_iter - 1).  (https://github.com/open-mmlab/mmengine/blob/main/mmengine/runner/loops.py#L314)

Suppose (save_steps = evaluation_freq = 5), we need to save eval results and ckpt when runner.iter equals to 4.